### PR TITLE
[skip] update example code in README

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

suggest using Init() in README

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [x] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] breaks backward compatibility
- [ ] requires a documentation update
- [ ] has untestable code

## Related issue/PR

- Closes #46 

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[major]`/`[minor]`/`[patch]`/`[skip]` in the PR title
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [x] Confirmed no dropping in test coverage (by [Codecov](https://codecov.io/gh/yahoojapan/athenz-authorizer/pulls))
- [x] Passed all pipeline checking
- [x] Approved by >1 reviewer

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[major]`/`[minor]`/`[patch]`/`[skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

**Review by scr-oath — APPROVED** [review]
Thanks!

File: README.md [hunk 1]
```diff
@@ -38,35 +37,61 @@ Athenz authorizer is a library to cache the policies of [Athenz](https://github.
...
+    // Initialize authorizerd
+    daemon, err := authorizerd.New(
+        authorizerd.WithAthenzURL("www.athenz.io"), // set athenz URL
+        authorizerd.WithAthenzDomains("domain1", "domain2", "domain N"), // set athenz domains
+        authorizerd.WithPubkeyRefreshPeriod("12h"), // optional, default: 24h
+        authorizerd.WithPolicyRefreshPeriod("1h"), // optional, default: 30m
+    )
+    if err != nil {
+        // cannot initialize authorizer daemon
+        log.Fatalf("daemon new error: %s\n", err.Error())
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply README.md]: fixed: 15fd059

File: README.md [hunk 2]
```diff
@@ -38,35 +37,61 @@ Athenz authorizer is a library to cache the policies of [Athenz](https://github.
...
+
+    // Verified results are returned
+    principal, err := daemon.AuthorizeRoleToken(ctx, roleTok, act, res)
+    if err != nil {
+        // token not authorized; take appropriate action
+        log.Fatalf("token not authorized: %s\n", err.Error())
+    }
+
+    // Inspect the authorized identity
+    log.Printf("authorized principal: %s\n", principal.Name())
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply README.md]: fixed: 15fd059

File: README.md [hunk 3]
```diff
@@ -38,35 +37,61 @@ Athenz authorizer is a library to cache the policies of [Athenz](https://github.
...
+    if err := daemon.VerifyRoleToken(ctx, roleTok, act, res); err != nil {
+        // token not authorized; take appropriate action
+        log.Fatalf("token not authorized: %s\n", err.Error())
+    }
+
+    // Verified results are returned
+    principal, err := daemon.AuthorizeRoleToken(ctx, roleTok, act, res)
+    if err != nil {
+        // token not authorized; take appropriate action
+        log.Fatalf("token not authorized: %s\n", err.Error())
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply README.md]: fixed: 15fd059

File: README.md [hunk 4]
```diff
@@ -38,35 +37,61 @@ Athenz authorizer is a library to cache the policies of [Athenz](https://github.
...
+    ctx := context.Background() // user can control authorizer daemon lifetime using this context
+    if err = daemon.Init(ctx); err != nil { // initialize internal daemons in dependency order (e.g. public keys before signed policies)
+        // cannot initialize internal daemons inside authorizer
+        log.Fatalf("daemon init error: %s\n", err.Error())
+    }
+    errs := daemon.Start(ctx)
+    go func() {
+        for err := range <-errs {
+            // user should handle errors return from the daemon
+            log.Printf("daemon start error: %s\n", err.Error())
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply README.md]: fixed: 15fd059

File: README.md [hunk 5]
```diff
@@ -38,35 +37,61 @@ Athenz authorizer is a library to cache the policies of [Athenz](https://github.
...
+        }
+    }()
+
+    roleTok := "<secret role token>"
+    act := "action"
+    res := "resource"
+
+    // Verify role token
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply README.md]: added access token example in 14e1273

File: README.md [hunk 6]
```diff
@@ -38,35 +37,61 @@ Athenz authorizer is a library to cache the policies of [Athenz](https://github.
...
+    if err := daemon.VerifyRoleToken(ctx, roleTok, act, res); err != nil {
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply README.md]: removed `verify*()` calls in 14e1273

**Review by scr-oath — APPROVED** [review]

**Review by falz-tino — APPROVED** [review]
LGTM

**Pull request merged**